grpc-server/grpc_server.go

```diff
@@ -236,7 +234,7 @@ func (s *Server) Subscribe(in *hegel.SubscribeRequest, stream hegel.Hegel_Subscr
 	)
 
 	// Disconnect previous client if a client is already connected for this hardware id
-	if old != nil {
+	if old := s.subscriptions[id]; old != nil {
```

Comment:
this is dereferencing subscriptions w/o locking it and is racy. Pretty sure the linter is wrong here.

Reply:
That's a great catch. Thank you (reverted)!